HPCC-26165 Retrieve dropzone files without specifying server

wangkx opened the pull request:
A dropzone server (hostname or IP) has to be specified when
retrieving dropzone files in the existing FileSpray service
(both DropZoneFileSearch and DropZoneFiles). By this fix,
if a dropzone server is not specified, the service returns
the matched files in all of the servers within a dropzone.

Also report the dropzone server for each dropzone file.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

HPCCSmoketest commented:
Process of PR-16153, label: h26165 starts now.
The reason of this test is: Forced to re-test.
Commit ID: 025ed7b104a12b0332e1b17b035c26c11f64ef5a
Estimated completion time is 0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-22-254-43.ec2.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.22.1,  cUrl: 7.67.0,  node.js: v16.13.1,  npm: 8.1.2

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-22-254-43.ec2.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.22.1,  cUrl: 7.67.0,  node.js: v16.13.1,  npm: 8.1.2
Sha: 025ed7b104a12b0332e1b17b035c26c11f64ef5a
Containerized:False
Build: success
Milestone:Install hpccsystems-platform-community_8.7.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |223 |223 |0 |0 |0 |66 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |9 |9 |0 |21 sec  (00:00:21) |
|setup (thor) |9 |9 |0 |38 sec  (00:00:38) |
|setup (roxie) |20 |20 |0 |18 sec  (00:00:18) |
|test (hthor) |986 |986 |0 |814 sec  (00:13:34) |
|test (thor) |896 |896 |0 |981 sec  (00:16:21) |
|test (roxie) |1056 |1056 |0 |925 sec  (00:15:25) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   35 sec (00:00:35) |  427 sec (00:07:07) |  104 sec (00:01:44) |   19 sec (00:00:19) |   15 sec (00:00:15) | 1242 sec (00:20:42) |   19 sec (00:00:19) | 1861 sec (00:31:01) |

afishbeck commented on esp/services/ws_fs/ws_fsService.cpp:
```diff
@@ -3165,7 +3163,7 @@ bool CFileSprayEx::onDropZoneFileSearch(IEspContext &context, IEspDropZoneFileSe
         unsigned filesFound = 0;
         IArrayOf<IEspPhysicalFileStruct> files;
-        bool isIPAddressReq = isIPAddress(dropZoneServerReq);
+        bool isIPAddressReq = !isEmptyString(dropZoneServerReq) && isIPAddress(dropZoneServerReq);
```
I think isIPAddress already checks for empty string?

wangkx replied:
The duplicated string check will be removed.

HPCCSmoketest commented:
Process of PR-16153, label: h26165 starts now.
The reason of this test is: Forced to re-test.
Commit ID: c2468fe2c5b9c281434005d67ef4168ddf108cba
Estimated completion time is 0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-22-254-239.ec2.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.22.1,  cUrl: 7.67.0,  node.js: v16.13.1,  npm: 8.1.2
VCPKG build disabled.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-22-254-239.ec2.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.22.1,  cUrl: 7.67.0,  node.js: v16.13.1,  npm: 8.1.2
Sha: c2468fe2c5b9c281434005d67ef4168ddf108cba
Containerized: False
VCPKG build disabled.
Build: success
Milestone:Install hpccsystems-platform-community_8.7.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |223 |223 |0 |0 |0 |68 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |0 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |9 |9 |0 |21 sec  (00:00:21) |
|setup (thor) |9 |9 |0 |38 sec  (00:00:38) |
|setup (roxie) |20 |20 |0 |18 sec  (00:00:18) |
|test (hthor) |986 |986 |0 |828 sec  (00:13:48) |
|test (thor) |896 |896 |0 |993 sec  (00:16:33) |
|test (roxie) |1056 |1056 |0 |932 sec  (00:15:32) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   33 sec (00:00:33) |  437 sec (00:07:17) |  125 sec (00:02:05) |   20 sec (00:00:20) |   18 sec (00:00:18) | 1257 sec (00:20:57) |   19 sec (00:00:19) | 1909 sec (00:31:49) |

ghalliday commented on esp/services/ws_fs/ws_fsService.cpp:
```diff
@@ -3357,14 +3354,36 @@ bool CFileSprayEx::getDropZoneFiles(IEspContext &context, const char* dropZone,
 }
 
-//This method returns all dropzones and, if NetAddress and Path specified, returns filtered list of files.
+void CFileSprayEx::getServersInDropZone(const char *dropZoneName, IArrayOf<IConstTpDropZone> &dropZoneList, bool ECLWatchVisibleOnly, StringArray &serverList)
```
trivial: unusual to have a variable that starts with a capital letter.

richardkchapman replied:
putting 'is' on the front may be a simple fix...

ghalliday reviewed (approved):
Looks reasonable to me

richardkchapman commented:
This should be squashed, and retargeted to 8.8.x

HPCCSmoketest commented:
Process of PR-16153, label: h26165 starts now.
The reason of this test is: New commit.
Commit ID: a3d5362ec1f9991448353c205f03ab77696cfa86
Estimated completion time is 0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-22-254-156.ec2.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.22.1,  cUrl: 7.67.0,  node.js: v16.13.1,  npm: 8.1.2
VCPKG build disabled.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-22-254-156.ec2.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.22.1,  cUrl: 7.67.0,  node.js: v16.13.1,  npm: 8.1.2
Sha: a3d5362ec1f9991448353c205f03ab77696cfa86
Containerized: False
VCPKG build disabled.
Build: success
Milestone:Install hpccsystems-platform-community_8.8.0-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |223 |223 |0 |0 |0 |67 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |9 |9 |0 |22 sec  (00:00:22) |
|setup (thor) |9 |9 |0 |38 sec  (00:00:38) |
|setup (roxie) |20 |20 |0 |18 sec  (00:00:18) |
|test (hthor) |986 |986 |0 |821 sec  (00:13:41) |
|test (thor) |896 |896 |0 |986 sec  (00:16:26) |
|test (roxie) |1056 |1056 |0 |926 sec  (00:15:26) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   30 sec (00:00:30) |  424 sec (00:07:04) |  105 sec (00:01:45) |   22 sec (00:00:22) |   17 sec (00:00:17) | 1248 sec (00:20:48) |   19 sec (00:00:19) | 1865 sec (00:31:05) |

wangkx commented:
@richardkchapman this PR has been squashed and retargeted to 8.8.x.

HPCCSmoketest commented:
Process of PR-16153, label: h26165 starts now.
The reason of this test is: Forced to re-test.
Commit ID: a3d5362ec1f9991448353c205f03ab77696cfa86
Estimated completion time is 0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-22-254-135.ec2.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.22.1,  cUrl: 7.67.0,  node.js: v16.13.1,  npm: 8.1.2
VCPKG build disabled.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-22-254-135.ec2.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.22.1,  cUrl: 7.67.0,  node.js: v16.13.1,  npm: 8.1.2
Sha: a3d5362ec1f9991448353c205f03ab77696cfa86
Containerized: False
VCPKG build disabled.
Build: success
Milestone:Install hpccsystems-platform-community_8.8.0-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |223 |223 |0 |0 |0 |66 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |9 |9 |0 |21 sec  (00:00:21) |
|setup (thor) |9 |9 |0 |38 sec  (00:00:38) |
|setup (roxie) |20 |20 |0 |19 sec  (00:00:19) |
|test (hthor) |986 |986 |0 |815 sec  (00:13:35) |
|test (thor) |896 |896 |0 |974 sec  (00:16:14) |
|test (roxie) |1056 |1056 |0 |868 sec  (00:14:28) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   35 sec (00:00:35) |  426 sec (00:07:06) |  104 sec (00:01:44) |   20 sec (00:00:20) |   19 sec (00:00:19) | 1236 sec (00:20:36) |   19 sec (00:00:19) | 1859 sec (00:30:59) |

richardkchapman commented:
There seem to be some unwanted submodule changes included after your rebase/retarget

wangkx commented:
Not sure why the 'git submodule update --init --recursive' did not clean that. I may need to manually remove it.

wangkx committed:
HPCC-26165 Retrieve dropzone files without specifying server
A dropzone server (hostname or IP) has to be specified when
retrieving dropzone files in the existing FileSpray service
(both DropZoneFileSearch and DropZoneFiles). By this fix,
if a dropzone server is not specified, the service returns
the matched files in all of the servers within a dropzone.

Also report the dropzone server for each dropzone file.

Revise based on review:
1. Remove duplicated IP Address string checks.
2. Remove the redundant return statement.
3. Rename the ECLWatchVisibleOnly to isECLWatchVisibleOnly.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

wangkx commented:
@richardkchapman the unwanted submodule changes have been removed.

HPCCSmoketest commented:
Process of PR-16153, label: h26165 starts now.
The reason of this test is: Forced to re-test.
Commit ID: 8928280d704ab8629cdc1d6d7fecc6fce737a65d
Estimated completion time is 0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-22-254-23.ec2.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.22.1,  cUrl: 7.67.0,  node.js: v16.13.1,  npm: 8.1.2
VCPKG build disabled.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-22-254-23.ec2.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.22.1,  cUrl: 7.67.0,  node.js: v16.13.1,  npm: 8.1.2
Sha: 8928280d704ab8629cdc1d6d7fecc6fce737a65d
Containerized: False
VCPKG build disabled.
Build: success
Milestone:Install hpccsystems-platform-community_8.8.0-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |223 |223 |0 |0 |0 |66 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |9 |9 |0 |21 sec  (00:00:21) |
|setup (thor) |9 |9 |0 |38 sec  (00:00:38) |
|setup (roxie) |20 |20 |0 |19 sec  (00:00:19) |
|test (hthor) |986 |986 |0 |812 sec  (00:13:32) |
|test (thor) |896 |896 |0 |982 sec  (00:16:22) |
|test (roxie) |1056 |1056 |0 |936 sec  (00:15:36) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   37 sec (00:00:37) |  426 sec (00:07:06) |  103 sec (00:01:43) |   19 sec (00:00:19) |   17 sec (00:00:17) | 1244 sec (00:20:44) |   19 sec (00:00:19) | 1865 sec (00:31:05) |
